src/Domain/Services/CartCheckoutPageFormat.php

```diff
@@ -0,0 +1,365 @@
+				$desired_checkout_format = isset( $_POST['cart_checkout_format_options_checkout_format'] ) ? sanitize_text_field( wp_unslash( $_POST['cart_checkout_format_options_checkout_format'] ) ) : '';
+
+				$info = self::get_cart_checkout_status();
+
+				if ( 'block' === $desired_cart_format && 'block' !== $info['cart_page_format'] ) {
+					self::set_cart_page_format_block(
+						$info['cart_page']
+					);
+				}
+
+				// Other cases coming soon - ( block | shortcode ) x ( cart | checkout ).
+			}
+		);
+	}
+
+	/**
+	 * Register hooks for admin pages.
+	 */
+	public function admin_init() {
+		add_filter(
+			'woocommerce_settings_pages',
+			function ( $pages ) {
+				$insert_after_index = array_search(
+					[
+						'type' => 'sectionend',
+						'id'   => 'checkout_process_options',
+					],
+					$pages,
+					true
+				);
+				if ( $insert_after_index ) {
+					// This includes a fudge factor to account for SSL items
+					// which don't have numeric indices.
+					// This could break if Woo core $pages content changes,
+					// or if other extensions filter/modify this index.
+					// If that happened, the items would be rendered in the
+					// wrong place in the settings page, so should only be a UX/
+					// cosmetic issue.
+					$insert_after_index = $insert_after_index + 3;
+				}
+				$insert_after_index    = min( count( $pages ), $insert_after_index );
+				$checkout_format_items = self::get_checkout_format_items();
+				array_splice( $pages, $insert_after_index, 0, $checkout_format_items );
+				return $pages;
+			}
+		);
+
+	}
+
+	/**
+	 * Generate an array of elements for displaying in `WooCommerce > Settings > Advanced` admin page.
+	 *
+	 * @return array Array of arrays, each element with various keys (e.g. title, type, id).
+	 */
+	private static function get_checkout_format_items() {
+		$block_status = self::get_cart_checkout_status();
+
+		$block_shortcode_options = [
+			'block'     => __( 'Block', 'woo-gutenberg-products-block' ),
+			'shortcode' => __( 'Shortcode', 'woo-gutenberg-products-block' ),
+			'custom'    => __( 'Custom', 'woo-gutenberg-products-block' ),
+		];
+
+		$items[] = [
+			'title' => __( 'Cart and checkout format', 'woo-gutenberg-products-block' ),
+			'type'  => 'title',
+
+			'desc'  => sprintf(
+				// Translators: %s Documentation url.
+				__( 'Choose between block or shortcode formats. <a href="%s">Learn more.</a>', 'woo-gutenberg-products-block' ),
+				'https://docs.woocommerce.com/document/cart-checkout-blocks-support-status/'
+			),
+
+			__( 'Choose between block or shortcode formats. Learn more.', 'woo-gutenberg-products-block' ),
+			'id'    => 'cart_checkout_format_options_title',
+		];
+
+		$description = 'Cart page is not set – select a page in the section above.';
+		if ( $block_status['cart_page_id'] ) {
```

Comment:
Notice if the page is not defined, its id is `-1`, so this evaluates to true.

Reply:
Good spotting, I think I meant to use `cart_page` (etc) here, as it's false if missing (or a page object if available). Fixed up here: 9782c5314d8864dec69a8393a30414d277401a61